just-the-docs/just-the-docs
# Fix navigation when `.html` omitted

Fix #1371

In `assets/js/just-the-docs.js`, function `navLink()` currently searches for navigation links whose `href` is the `pathname` of the page either with or without a terminating `/`. However, it doesn't ignore a terminating `.html`.

To reproduce issue #1371 on the theme docs site:

1. Remove `permalink: pretty` from `_config.yml`.
2. Build and browse the theme docs with the current release.
3. Select `UI Components: Buttons` in the navigation panel.
4. Check that the URL ends with `/buttons.html`.
5. Remove `.html` from the URL.
6. Check that the navigation for `UI Components` is no longer expanded, so the link to `Buttons` isn't shown.

This PR enhances `navLink()` to ignore a terminating `.html` when the `pathname` does not match exactly:
- Remove `.html` from `pathname`
- Find nav link whose `href` is `pathname` with or without `.html`

### Testing

Remove `permalink: pretty` from `_config.yml`.

1. Build and browse the theme docs locally.
2. Select `UI Components: Typography` in the navigation panel.
3. Check that the navigation for `UI Components` is expanded.
4. Check that the page URL ends with `typography.html`.
5. Replace `typography.html` by `buttons` in the page URL.
6. Check that the navigation for `UI Components` remains expanded, with `Buttons` highlighted.
7. Repeat steps 2–6  with `buttons/` instead of `buttons`.

Restore `permalink: pretty` in `_config.yml`.

1. Build and browse the theme docs locally.
2. Select `UI Components: Typography` in the navigation panel.
3. Check that the navigation for `UI Components` is expanded.
4. Check that the page URL ends with `typography/`.
5. Replace `typography/` by `buttons` in the page URL.
6. Check that the navigation for `UI Components` remains expanded, with `Buttons` highlighted.

### Caveats

- I have very little experience of writing JS.
- I don't know much about deployment, but it appears that not all servers support dropping `.html` from URLs. Could this change cause problems for users who deploy JTD sites to other servers than GitHub Pages?
